js/capture.js

```diff
@@ -19,13 +27,61 @@ const capture = {
       });
   },
 
+  // In the parsed whitelist JSON object (from disconnect-entitylist.json):
+  //   - 'properties' keys are first parties
+  //   - 'resources' keys are third parties
+  reformatList(whiteList) {
+    const firstPartyWhiteList = {};
+    const thirdPartyWhiteList = {};
+    let counter = 0;
+    for (const siteOwner in whiteList) {
+      const firstParties = whiteList[siteOwner].properties;
+      for (let i = 0; i < firstParties.length; i++) {
+        firstPartyWhiteList[firstParties[i]] = counter;
+      }
+      const thirdParties = whiteList[siteOwner].resources;
+      thirdPartyWhiteList[counter] = [];
+      for (let i = 0; i < thirdParties.length; i++) {
+        thirdPartyWhiteList[counter].push(thirdParties[i]);
+      }
+      counter++;
+    }
+
+    return {
+      firstPartyWhiteList,
+      thirdPartyWhiteList
+    };
+  },
+
   // Returns true if the request should be stored, otherwise false.
-  shouldStore(tab) {
+  shouldStore(tab, thirdPartyFromRequest) {
     const documentUrl = new URL(tab.url);
     // ignore about:*, moz-extension:* & non-visible tabs (like dev tools)
     if (documentUrl.protocol !== 'about:'
       && documentUrl.protocol !== 'moz-extension:'
       && tab.id !== browser.tabs.TAB_ID_NONE) {
+      // check if third party is on the whitelist (owned by the first party)
+      if (thirdPartyFromRequest) {
+        const firstPartyFromRequest = documentUrl.hostname;
+        for (const firstParty in this.firstPartyWhiteList) {
+          const firstPartyFromList = new RegExp(firstParty);
```

Comment:
> Hmm well if you are just string matching why can't you just look up the thirdParty object key rather than comparing? Maybe the issue is also your data structure... let me check.

Do you mean why can't I use something like `whitelist.hasOwnProperty(firstParty)`? If so, it's because the first party string from the request is often not a direct match for what ought to be the corresponding string/property in the transformed whitelist object.

> Such that google.com would match www.google.com and blah.google.com. I'm not exactly sure if we want to do this for anything other than www. variants. 

But blah.google.com is google.com, so shouldn't it be subject to the same whitelist ignore check? Why would we only be interested in `www`?

> The problem with this RegExp also is that it would also make google.com match for google.com.arse.shoe.com also which is an issue.

I can modify the regExp to only look for cases where the substring `google.com` occurs at the end of the string in question. Would that solve it?